fusionfusion/run.py

```diff
@@ -134,18 +138,69 @@ def fusionfusion_main(args):
     ####################
     # parsing chimeric reads from bam files
     if starBamFile is not None:
-
-        parseJunctionInfo.parseJuncInfo_STAR(starBamFile, output_dir + "/star.chimeric.tmp.txt")
-                                             
-
-        hOUT = open(output_dir + "/star.chimeric.txt", "w")
-        subprocess.check_call(["sort", "-k1,1", "-k2,2n", "-k4,4", "-k5,5n", output_dir + "/star.chimeric.tmp.txt"], stdout = hOUT)
-        hOUT.close()
-
-        cluster_filter_junction(output_dir + "/star.chimeric.txt", output_dir + "/star", args)
+        generates_trace = False
+        starSjTab = args.star_sj_tab
+        starAlignedBam = args.star_aligned_bam
+        if starSjTab is None or starAlignedBam is None:
+            if starSjTab is not None or starAlignedBam is not None:
+                msg = "To enable further analysis using STAR's SJ.out.tab, both --star_sj_tab and --star_aligned_bam options must be specified."
+                print(msg, file=sys.stderr)
+
+            parseJunctionInfo.parseJuncInfo_STAR(starBamFile, output_dir + "/star.chimeric.tmp.txt")
+
+            with open(output_dir + "/star.chimeric.txt", "w") as hOUT:
+                subprocess.check_call([
+                    "sort", "-k1,1", "-k2,2n", "-k4,4", "-k5,5n",
+                    output_dir + "/star.chimeric.tmp.txt"
+                ], stdout=hOUT)
+        else:
+            parseJunctionInfo.parseJuncInfo_STAR(starBamFile,
+                                                 output_dir + "/star.chimeric.tmp1.txt",
+                                                 source="Chimeric")
+
+            genome_id = args.genome_id
+            is_grc = args.grc
+            annot.gene.make_gene_info(output_dir + "/star.tmp.refGene.bed.gz", "refseq", genome_id, is_grc, False)
+            annot.gene.make_gene_info(output_dir + "/star.tmp.ensGene.bed.gz", "gencode", genome_id, is_grc, False)
+
+            with pysam.TabixFile(output_dir + "/star.tmp.refGene.bed.gz") as ref_gene_tb, \
+                pysam.TabixFile(output_dir + "/star.tmp.ensGene.bed.gz") as ens_gene_tb:
+                filt = ShortRangeChimeraFilter(ref_gene_tb, ens_gene_tb)
+                filt.run(starSjTab, starAlignedBam, output_dir + "/star.chimeric.tmp2.sam")
+            subprocess.check_call(["rm", output_dir + "/star.tmp.refGene.bed.gz"])
+            subprocess.check_call(["rm", output_dir + "/star.tmp.ensGene.bed.gz"])
+            subprocess.check_call(["rm", output_dir + "/star.tmp.refGene.bed.gz.tbi"])
+            subprocess.check_call(["rm", output_dir + "/star.tmp.ensGene.bed.gz.tbi"])
```

Comment:
Use the standard lib to avoid forking

```suggestion
            os.remove(output_dir + "/star.tmp.refGene.bed.gz")
            os.remove(output_dir + "/star.tmp.ensGene.bed.gz")
            os.remove(output_dir + "/star.tmp.refGene.bed.gz.tbi")
            os.remove(output_dir + "/star.tmp.ensGene.bed.gz.tbi")
```

or, use `tempfile` to remove them automatically